File: src/types/diagnostics.ts
// 诊断相关类型，描述设置中心连通性检测结果。

export interface DiagnosticsConnectivityItem {
  id: string;
  label: string;
  target: string;
  status: 'reachable' | 'failed' | 'skipped' | string;
  latency_ms?: number | null;
  message: string;
}

export interface DiagnosticsConnectivityResponse {
  checks: DiagnosticsConnectivityItem[];
}

File: crates/muse-local-api/src/dto/diagnostics.rs
//! 诊断接口的响应 DTO。

use serde::Serialize;

/// 单项连通性诊断结果。
#[derive(Serialize)]
pub struct DiagnosticsConnectivityItem {
    pub id: String,
    pub label: String,
    pub target: String,
    pub status: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub latency_ms: Option<u64>,
    pub message: String,
}

/// 连通性诊断响应体。
#[derive(Serialize)]
pub struct DiagnosticsConnectivityResponse {
    pub checks: Vec<DiagnosticsConnectivityItem>,
}
